Recreate PR for ShareIt v2.0 add-controllers

Zazergel opened the pull request:
Пересоздал пул реквест, в предыдущем, почему-то, гитхаб не стал проверять мой коммит(

Zazergel committed:
ShareIt v1.0 add-controllers
Реализовал базу приложения

Zazergel committed:
ShareIt v1.1 add-controllers
fix: code-style #1

Zazergel committed:
ShareIt v2.0 add-controllers
feat: Полностью переработана логика приложения (почти что с нуля), теперь операции с моделями берут на себя хранилища (валидация, создание, хранение, изменение, поиск, присвоение id и т.д.)
fix: Переработаны аннотации в классах - моделях.
feat: Дополнены "мапперы" моделей.
fix: Теперь контроллеры оперируют Dto объектами, а не моделями.
fix: Немного переработан ErrorHandler
fix: исправлена опечатка поля "broker" в классе Booking.java на "bookerId"
fix: code-style #2
refactor: пакеты, содержащие всего один класс были удалены, классы извлечены в пакет повыше.
delete: удалены лишние методы equals & hashcode в классах - моделях
delete: удалены лишние аннотации @Data.

Zazergel committed:
ShareIt v2.1 add-controllers (повторный коммит)
feat: добавлен 1 тестовый комментарий

NikolaiDem commented on src/main/java/ru/practicum/shareit/item/dto/ItemMapper.java:
```diff
@@ -0,0 +1,31 @@
+import org.springframework.stereotype.Component;
+import ru.practicum.shareit.item.Item;
+
+@Component
```
Делать бином не обязательно, так как методы статические

NikolaiDem commented on src/main/java/ru/practicum/shareit/item/service/ItemServiceImpl.java:
```diff
@@ -0,0 +1,79 @@
+@Slf4j
+public class ItemServiceImpl implements ItemService {
+    private final ItemStorage itemStorage;
+    private final ItemMapper itemMapper;
```
Не используется

NikolaiDem commented on src/main/java/ru/practicum/shareit/item/storage/ItemStorageInMemory.java:
```diff
@@ -0,0 +1,82 @@
+
+    @Override
+    public Item addItem(Item item) {
+        userService.get(item.getOwnerId());
```
Валидацию лучше делать в сервисе. Задача репозитория - CRUD операции

NikolaiDem commented on src/main/java/ru/practicum/shareit/user/dto/UserMapper.java:
```diff
@@ -0,0 +1,23 @@
+import org.springframework.stereotype.Component;
+import ru.practicum.shareit.user.User;
+
+@Component
```
Делать бином не обязательно, так как методы статические

NikolaiDem commented on src/main/java/ru/practicum/shareit/user/service/UserServiceImpl.java:
```diff
@@ -0,0 +1,52 @@
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserStorage userStorage;
+    private final UserMapper userMapper;
```
Не используется

Zazergel committed:
ShareIt v2.2 add-controllers
refactor: валидация на наличие пользователя перенесена из хранилища в сервис.
delete: Убраны лишние аннотации Component в классах - мапперах
delete: Убраны не использующиеся поля - мапперы из классов Сервисов.
